### ISSUE TITLE
Significant slowdown in Keras model.fit() on simple problem when using validation data

### ISSUE THREAD

**Issue by aaustin141** [tensorflow/tensorflow#58828]
<details><summary>Click to expand!</summary> 
 
 ### Issue Type

Performance

### Source

binary

### Tensorflow Version

2.11.0

### Custom Code

No

### OS Platform and Distribution

Ubuntu 18.04

### Python version

3.10.8

### CUDA/cuDNN version

11.2 / 8.1.0

### GPU model and memory

NVIDIA Quadro GV100 (32 GB)

### Current Behaviour?

I observe a significant slowdown when using Keras model.fit() to train a simple
3-layer fully-connected network when the validation_data parameter is set.  A
typical run of the provided code on my machine results in the following output:
```
Time with validation:     3.89 s
Time without validation:  0.39 s
```
The validation data set is 1/8 the size of the training dataset.  I don't
expect validation to be free, but it should not result in a 10x slowdown.

Some notes / things I've tried:

 - Setting validation_batch_size makes no difference.

 - Using a single NumPy array to hold both the training and validation datasets
   and using validation_split instead of validation_data also makes no
   difference.

 - I found the possibly related issue #57645, which suggests that it might be a
   problem related to eager execution.  The poster of that issue is able to fix
   things by calling tf.compat.v1.disable_eager_execution(), but if I try this
   in my own code with TF 2.11.0, I run into errors (not really surprising).

 - One of the respondents on that same issue suggested swapping out the NumPy
   arrays for tf.data.Dataset objects.  I found this made no difference, even
   using `.prefetch(tf.data.AUTOTUNE)` as recommended.

 - Deactivating the GPU (by setting CUDA_VISIBLE_DEVICES=-1) makes no
   difference to the time with validation but increases the time without
   validation by about 50%.  That suggests the validation step might not be
   leveraging the GPU, which would certainly be an issue, but that's still
   not enough to account for a 6x slowdown.

Is this a bug?  Or am I missing something obvious?

### Standalone code to reproduce the issue

```python
import tensorflow as tf
import numpy as np
import time
    
xTrain = np.random.randn(2048, 128)
yTrain = np.random.randn(2048, 128)
xValid = np.random.randn(256,  128)
yValid = np.random.randn(256,  128)
                
model = tf.keras.models.Sequential([
    tf.keras.layers.Input(shape = (128,)),
    tf.keras.layers.Dense(128, activation = "tanh"),
    tf.keras.layers.Dense(128)
])      
        
model.compile(
    optimizer = tf.keras.optimizers.Adam(1.0e-3),
    loss = tf.keras.losses.MeanSquaredError(),
)   
    
# Preliminary run to build graphs.
model.fit(x = xTrain, y = yTrain, epochs = 5, batch_size = 2048,
    validation_data = (xValid, yValid)
)

t1 = time.time()

model.fit(x = xTrain, y = yTrain, epochs = 100, batch_size = 2048,
    validation_data = (xValid, yValid)
)

t2 = time.time()

model.fit(x = xTrain, y = yTrain, epochs = 100, batch_size = 2048)

t3 = time.time()

print("Time with validation:     %.2f s" % (t2 - t1))
print("Time without validation:  %.2f s" % (t3 - t2))
```

### Relevant log output

_No response_</details>

**Comment by mohantym** [comment]
Hi @aaustin141 !
I was able to replicate this issue in [Colab(](https://colab.sandbox.google.com/gist/mohantym/7381c0c31c7ed3ed0bffb60361d5681f/git_58828.ipynb#scrollTo=_gjrJV7P84oN) approx 3.5 times slow in 2.11 version).

1/10 inference time compared to  inference time without dataset might be valid issue.

But  3.5 times in validation dataset looks reasonable compared to training without validation dataset.

Please post this issue on [keras-team/keras repo](https://github.com/keras-team/keras/issues)
To know more see;
[https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999](https://discuss.tensorflow.org/t/keras-project-moved-to-new-repository-in-https-github-com-keras-team-keras/1999) . 

Thank you!

**Comment by mohantym** [comment]
Ok @aaustin141 !

Thanks for the clarification. 
Moving this issue to closed status as it will be tracked in [#17328](https://github.com/keras-team/keras/issues/17328).

Thank you!
